### PR TITLE
[misc] Fix some bugs, editorconfig, isort and black

### PR DESCRIPTION
**Pull request by apsdehal**
- See individual commits for more details
- Synchro and save checkpoint only in main process
- Fix a bug in early stopping
- Black and isort
- Add editorconfig for better cross editor compatibility
- Absolute imports instead of relative
- Logger now supports log all to log in all processes in distributed setting

### CODE REVIEW

File: pythia/common/trainer.py [hunk 1]
```diff
@@ -128,12 +128,13 @@ def load_model(self):
...
-            self.writer.write(
-                "CUDA Device {} is: {}".format(
-                    rank, torch.cuda.get_device_name(self.local_rank)
-                )
+            device_info = "CUDA Device {} is: {}".format(
+                rank, torch.cuda.get_device_name(self.local_rank)
             )
 
+            self.writer.write(device_info)
+            print(device_info)
```
Comment (abhiskk) [inline pythia/common/trainer.py]: NIT: If possible use logging here instead of print statements.

File: pythia/common/trainer.py [hunk 2]
```diff
@@ -351,14 +351,16 @@ def _logistics(self, report):
...
 
         return should_break
 
     def _try_full_validation(self, force=False):
+        should_break = False
+
         if self.current_iteration % self.snapshot_interval == 0 or force:
-            self.writer.write("Evaluation time. Running on full "
-                              "validation set...")
+            self.writer.write("Evaluation time. Running on full " "validation set...")
```
Comment (dexter1691) [inline pythia/common/trainer.py]: nit: remove the two extra quotes here

Comment (apsdehal) [reply pythia/common/trainer.py]: Happens when I run black. Thanks.

File: pythia/common/trainer.py (hunk 1 above)
Comment (apsdehal) [reply pythia/common/trainer.py]: Will be fixed, requires some changes in logger.

**Pull request merged**